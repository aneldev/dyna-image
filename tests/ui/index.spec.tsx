import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import * as React from 'react';
import * as enzyme from 'enzyme';

import {
  DynaImage,
  EImageMode,
} from "../../src";

configure({adapter: new Adapter()});

describe('Home', () => {
  it('rendered with minimum set of props', () => {
    const wrapper = enzyme.shallow(
      (
        <DynaImage src="https://www.example.com/eifel.jpg"/>
      ),
      {},
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('rendered with all possible props', () => {
    const wrapper = enzyme.shallow(
      (
        <DynaImage
          className="className"
          src="https://www.example.com/eifel.jpg"
          mode={EImageMode.FILL}
          alt="Profile photo"
          onLoad={() => undefined}
          onError={() => undefined}
        />
      ),
      {},
    );
    expect(wrapper).toMatchSnapshot();
  });
});
