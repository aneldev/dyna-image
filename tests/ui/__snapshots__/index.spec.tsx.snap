// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendered with all possible props 1`] = `
<Fragment>
  <Loading
    className="className"
    isLoading={false}
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://www.example.com/eifel.jpg)",
          "backgroundSize": "cover",
        }
      }
    />
  </Loading>
  <img
    alt="Profile photo"
    hidden={true}
    onError={[Function]}
    onLoad={[Function]}
    src="https://www.example.com/eifel.jpg"
  />
</Fragment>
`;

exports[`Home rendered with minimum set of props 1`] = `
<Fragment>
  <Loading
    className=""
    isLoading={false}
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://www.example.com/eifel.jpg)",
          "backgroundSize": "contain",
        }
      }
    />
  </Loading>
  <img
    hidden={true}
    onError={[Function]}
    onLoad={[Function]}
    src="https://www.example.com/eifel.jpg"
  />
</Fragment>
`;
